app/objectiveReview/get.controller.js

```diff
@@ -7,68 +7,76 @@ const {
-  const objective = await getSentencePlanObjective(planId, objectiveId, token).catch(error =>
-    renderError(`Could not retrieve objective ${objectiveId} for sentence plan ${planId}`, error, res)
-  )
-  const motivations = await getMotivations(token).catch(error =>
-    renderError(`Could not retrieve motivation list`, error, res)
-  )
-
-  // only get needs data if there is an action with needs
-  if (objective.needs && objective.needs.length > 0) {
-    const needs = await getSentencePlanNeeds(planId, token).catch(error =>
-      renderError(`Could not retrieve needs for sentence plan ${planId}`, error, res)
+  try {
+    const {
+      path,
+      errors,
+      errorSummary,
+      params: { planId, objectiveId },
+      headers: { 'x-auth-token': token },
+    } = req
+    const nexturl = path.substring(0, path.lastIndexOf('/'))
+    const backurl = removeUrlLevels(path, 3)
+    const objective = await getSentencePlanObjective(planId, objectiveId, token).catch(error =>
+      catchAndReThrowError(`Could not retrieve objective ${objectiveId} for sentence plan ${planId}`, error)
     )
-    objective.needs = objective.needs.map(need => needs.find(({ id }) => id === need).name)
-  }
-
-  // only get intervention data if there is an action with an intervention
-  const hasInterventions = !objective.actions.every(({ intervention }) => !intervention)
-  let interventions = []
-  if (hasInterventions) {
-    interventions = await getInterventions(token).catch(error =>
-      renderError(`Could not retrieve objective ${objectiveId} for sentence plan ${planId}`, error, res)
+    const motivations = await getMotivations(token).catch(error =>
+      catchAndReThrowError(`Could not retrieve motivation list`, error)
     )
-  }
-  objective.actions = objective.actions.map(action => {
-    const { description, intervention, motivationUUID, owner, ownerOther, targetDate, status } = action
-    const interventionText =
-      intervention && `Intervension: ${interventions.find(({ uuid }) => uuid === intervention).shortDescription}`
-    const { monthName, year } = getYearMonthFromDate(targetDate)
-    const ownerText = owner.map(ownerType => RESPONSIBLE_LIST.find(({ value }) => value === ownerType).text).join(', ')
-    const ownerOtherText = ownerOther ? `: ${ownerOther}` : ''
-    return {
-      description: interventionText || description,
-      motivation: motivations.find(({ uuid }) => uuid === motivationUUID).motivationText,
-      targetDate: `${monthName} ${year}`,
-      owner: `${ownerText}${ownerOtherText}`,
-      status: getStatusText(status),
+    // only get needs data if there is an action with needs
+    if (objective.needs && objective.needs.length > 0) {
+      const needs = await getSentencePlanNeeds(planId, token).catch(error =>
+        catchAndReThrowError(`Could not retrieve needs for sentence plan ${planId}`, error)
+      )
+      objective.needs = objective.needs.map(need => needs.find(({ id }) => id === need).name)
```

Comment:
Just need to check the logic here - a need remains attached to an objective even if it is later not in the list of retrieved needs for the plan.